File: PlacementCellBackend/Controllers/PlacementAnalyticsController.cs
using Microsoft.AspNetCore.Mvc;
using PlacementCellBackend.DTOs;
using PlacementCellBackend.Services.Interfaces;

namespace PlacementCellBackend.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class PlacementAnalyticsController : ControllerBase
    {
        private readonly IPlacementAnalyticsService _analyticsService;

        public PlacementAnalyticsController(IPlacementAnalyticsService analyticsService)
        {
            _analyticsService = analyticsService;
        }

        /// <summary>
        /// Get comprehensive dashboard statistics
        /// </summary>
        [HttpGet("dashboard")]
        public async Task<ActionResult<DashboardStats>> GetDashboardStats()
        {
            var stats = await _analyticsService.GetDashboardStatsAsync();
            return Ok(stats);
        }

        /// <summary>
        /// Get top hiring companies ranking
        /// </summary>
        [HttpGet("companies/top/{count:int?}")]
        public async Task<ActionResult<IEnumerable<CompanyRanking>>> GetTopCompanies(int count = 10)
        {
            var rankings = await _analyticsService.GetTopCompaniesRankingAsync(count);
            return Ok(rankings);
        }

        /// <summary>
        /// Get detailed insights for a specific company
        /// </summary>
        [HttpGet("companies/{companyId}/insights")]
        public async Task<ActionResult<CompanyInsights>> GetCompanyInsights(string companyId)
        {
            var insights = await _analyticsService.GetCompanyInsightsAsync(companyId);
            return Ok(insights);
        }

        /// <summary>
        /// Get alumni statistics
        /// </summary>
        [HttpGet("alumni/statistics")]
        public async Task<ActionResult<AlumniStatistics>> GetAlumniStatistics()
        {
            var stats = await _analyticsService.GetAlumniStatisticsAsync();
            return Ok(stats);
        }

        /// <summary>
        /// Get recent placements
        /// </summary>
        [HttpGet("placements/recent/{count:int?}")]
        public async Task<ActionResult<IEnumerable<RecentPlacement>>> GetRecentPlacements(int count = 10)
        {
            var placements = await _analyticsService.GetRecentPlacementsAsync(count);
            return Ok(placements);
        }

        /// <summary>
        /// Get placement trends
        /// </summary>
        [HttpGet("trends")]
        public async Task<ActionResult<PlacementTrends>> GetPlacementTrends()
        {
            var trends = await _analyticsService.GetPlacementTrendsAsync();
            return Ok(trends);
        }

        /// <summary>
        /// Get industry distribution of placements
        /// </summary>
        [HttpGet("industries")]
        public async Task<ActionResult<IEnumerable<IndustryDistribution>>> GetIndustryDistribution()
        {
            var distribution = await _analyticsService.GetIndustryDistributionAsync();
            return Ok(distribution);
        }

        /// <summary>
        /// Get interview insights for a company
        /// </summary>
        [HttpGet("interviews/{companyId}")]
        public async Task<ActionResult<InterviewInsights>> GetInterviewInsights(string companyId)
        {
            var insights = await _analyticsService.GetInterviewInsightsAsync(companyId);
            return Ok(insights);
        }
    }
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/controllers/PlacementAnalyticsController.java
package com.example.placementicsbackend.controllers;

import com.example.placementicsbackend.dtos.AlumniStatistics;
import com.example.placementicsbackend.dtos.CompanyInsights;
import com.example.placementicsbackend.dtos.CompanyRanking;
import com.example.placementicsbackend.dtos.DashboardStats;
import com.example.placementicsbackend.dtos.PlacementTrends;
import com.example.placementicsbackend.dtos.RecentPlacement;
import com.example.placementicsbackend.dtos.RecommendedResources;
import com.example.placementicsbackend.services.analytics.interfaces.IPlacementAnalyticsService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/placement-analytics")
@RequiredArgsConstructor
public class PlacementAnalyticsController {

    private final IPlacementAnalyticsService analyticsService;

    @GetMapping("/dashboard")
    public ResponseEntity<DashboardStats> getDashboardStats() {
        return ResponseEntity.ok(analyticsService.getDashboardStats());
    }

    @GetMapping("/companies/top/{count}")
    public ResponseEntity<List<CompanyRanking>> getTopCompanies(@PathVariable int count) {
        return ResponseEntity.ok(analyticsService.getTopCompaniesRanking(count));
    }

    @GetMapping("/companies/top")
    public ResponseEntity<List<CompanyRanking>> getTopCompanies() {
        return ResponseEntity.ok(analyticsService.getTopCompaniesRanking(10));
    }

    @GetMapping("/companies/{companyId}/insights")
    public ResponseEntity<CompanyInsights> getCompanyInsights(@PathVariable String companyId) {
        return ResponseEntity.ok(analyticsService.getCompanyInsights(companyId));
    }

    @GetMapping("/alumni/statistics")
    public ResponseEntity<AlumniStatistics> getAlumniStatistics() {
        return ResponseEntity.ok(analyticsService.getAlumniStatistics());
    }

    @GetMapping("/placements/recent/{count}")
    public ResponseEntity<List<RecentPlacement>> getRecentPlacements(@PathVariable int count) {
        return ResponseEntity.ok(analyticsService.getRecentPlacements(count));
    }

    @GetMapping("/placements/recent")
    public ResponseEntity<List<RecentPlacement>> getRecentPlacements() {
        return ResponseEntity.ok(analyticsService.getRecentPlacements(10));
    }

    @GetMapping("/trends")
    public ResponseEntity<PlacementTrends> getPlacementTrends() {
        return ResponseEntity.ok(analyticsService.getPlacementTrends());
    }

    @GetMapping("/resources/recommended/{count}")
    public ResponseEntity<RecommendedResources> getTopRecommendedResources(@PathVariable int count) {
        return ResponseEntity.ok(analyticsService.getTopRecommendedResources(count));
    }

    @GetMapping("/resources/recommended")
    public ResponseEntity<RecommendedResources> getTopRecommendedResources() {
        return ResponseEntity.ok(analyticsService.getTopRecommendedResources(10));
    }
}
